Installation issue: multiple 'march' flags injected

### Steps to reproduce

I ran ```spack --debug install kokkos%gcc@12.3.0``` to install kokkos using gcc 12.3.0 compiler.

Output of `spack spec`:

```$ spack spec kokkos%gcc@12.3.0
[+]  kokkos@4.6.01~aggressive_vectorization~cmake_lang~compiler_warnings+complex_align~cuda~debug~debug_bounds_check~debug_dualview_modify_check~deprecated_code~examples~hip_relocatable_device_code~hpx~hpx_async_dispatch~hwloc~ipo~memkind~numactl~openmp~openmptarget~pic~rocm+serial+shared~sycl~tests~threads~tuning~wrapper build_system=cmake build_type=Release cxxstd=17 generator=make intel_gpu_arch=none arch=linux-ubuntu22.04-skylake
[+]      ^cmake@3.31.6~doc+ncurses+ownlibs~qtgui build_system=generic build_type=Release arch=linux-ubuntu22.04-skylake
[+]          ^curl@8.11.1~gssapi~ldap~libidn2~librtmp~libssh~libssh2+nghttp2 build_system=autotools libs:=shared,static tls:=openssl arch=linux-ubuntu22.04-skylake
[+]              ^nghttp2@1.65.0 build_system=autotools arch=linux-ubuntu22.04-skylake
[+]                  ^diffutils@3.10 build_system=autotools arch=linux-ubuntu22.04-skylake
[+]                      ^libiconv@1.18 build_system=autotools libs:=shared,static arch=linux-ubuntu22.04-skylake
[+]              ^openssl@3.4.1~docs+shared build_system=generic certs=mozilla arch=linux-ubuntu22.04-skylake
[+]                  ^ca-certificates-mozilla@2025-02-25 build_system=generic arch=linux-ubuntu22.04-skylake
[+]                  ^perl@5.40.0+cpanm+opcode+open+shared+threads build_system=generic arch=linux-ubuntu22.04-skylake
[+]                      ^berkeley-db@18.1.40+cxx~docs+stl build_system=autotools patches:=26090f4,b231fcc arch=linux-ubuntu22.04-skylake
[+]                      ^bzip2@1.0.8~debug~pic+shared build_system=generic arch=linux-ubuntu22.04-skylake
[+]                      ^gdbm@1.23 build_system=autotools arch=linux-ubuntu22.04-skylake
[+]                          ^readline@8.2 build_system=autotools patches:=1ea4349,24f587b,3d9885e,5911a5b,622ba38,6c8adf8,758e2ec,79572ee,a177edc,bbf97f1,c7b45ff,e0013d9,e065038 arch=linux-ubuntu22.04-skylake
[+]              ^pkgconf@2.3.0 build_system=autotools arch=linux-ubuntu22.04-skylake
[+]          ^ncurses@6.5~symlinks+termlib abi=none build_system=autotools patches:=7a351bc arch=linux-ubuntu22.04-skylake
[+]          ^zlib-ng@2.2.4+compat+new_strategies+opt+pic+shared build_system=autotools arch=linux-ubuntu22.04-skylake
[+]      ^compiler-wrapper@1.0 build_system=generic arch=linux-ubuntu22.04-skylake
[e]      ^gcc@12.3.0~binutils+bootstrap~graphite~mold~nvptx~piclibs~profiled~strip build_system=autotools build_type=RelWithDebInfo languages:='c,c++,fortran' arch=linux-ubuntu22.04-skylake
[+]      ^gcc-runtime@12.3.0 build_system=generic arch=linux-ubuntu22.04-skylake
[e]      ^glibc@2.35 build_system=autotools arch=linux-ubuntu22.04-skylake
[+]      ^gmake@4.4.1~guile build_system=generic arch=linux-ubuntu22.04-skylake
```

Maintainers:
@cedricchevalier19 @lucbv @nmm0 

Issue: 

First 15 lines of the file `spack-cc-kokkos-xl7f3yf.in.log`:

```
[ccld] /usr/bin/g++-12 CMakeCXXCompilerId.cpp
[ccld] /usr/bin/g++-12 -print-sysroot
[ld] ld --help
[vcheck] /usr/bin/g++-12 
[vcheck] /usr/bin/g++-12 
[ld] ld --help
[ld] ld --help
[cc] /usr/bin/g++-12 -std=gnu++17 -o CMakeFiles/cmTC_08b48.dir/cplusplus17.cpp.o -c /tmp/er12izuv/spack-stage/spack-stage-kokkos-4.6.01-xl7f3yfrba5idkjal25deocxj44elpbk/spack-src/cmake/compile_tests/cplusplus17.cpp
[ccld] /usr/bin/g++-12 CMakeFiles/cmTC_08b48.dir/cplusplus17.cpp.o -o cmTC_08b48
[ld] ld --help
[cc] /usr/bin/g++-12 -std=gnu++17 -o CMakeFiles/cmTC_1da0e.dir/CheckIncludeFile.cxx.o -c /tmp/er12izuv/spack-stage/spack-stage-kokkos-4.6.01-xl7f3yfrba5idkjal25deocxj44elpbk/spack-build-xl7f3yf/CMakeFiles/CMakeScratch/TryCompile-S1UePn/CheckIncludeFile.cxx
[ld] ld --help
[cc] /usr/bin/g++-12 -std=gnu++17 -o CMakeFiles/cmTC_90bad.dir/CheckIncludeFile.cxx.o -c /tmp/er12izuv/spack-stage/spack-stage-kokkos-4.6.01-xl7f3yfrba5idkjal25deocxj44elpbk/spack-build-xl7f3yf/CMakeFiles/CMakeScratch/TryCompile-tuSJ2G/CheckIncludeFile.cxx
[cc] /usr/bin/g++-12 -Dkokkossimd_EXPORTS -I/tmp/er12izuv/spack-stage/spack-stage-kokkos-4.6.01-xl7f3yfrba5idkjal25deocxj44elpbk/spack-build-xl7f3yf/simd/src -I/tmp/er12izuv/spack-stage/spack-stage-kokkos-4.6.01-xl7f3yfrba5idkjal25deocxj44elpbk/spack-src/simd/src -I/tmp/er12izuv/spack-stage/spack-stage-kokkos-4.6.01-xl7f3yfrba5idkjal25deocxj44elpbk/spack-build-xl7f3yf -O3 -DNDEBUG -std=gnu++17 -fPIC -MD -MT simd/src/CMakeFiles/kokkossimd.dir/Kokkos_SIMD_dummy.cpp.o -MF CMakeFiles/kokkossimd.dir/Kokkos_SIMD_dummy.cpp.o.d -o CMakeFiles/kokkossimd.dir/Kokkos_SIMD_dummy.cpp.o -c /tmp/er12izuv/spack-stage/spack-stage-kokkos-4.6.01-xl7f3yfrba5idkjal25deocxj44elpbk/spack-src/simd/src/Kokkos_SIMD_dummy.cpp
[cc] /usr/bin/g++-12 -DKOKKOS_DEPENDENCE -Dkokkoscore_EXPORTS -I/tmp/er12izuv/spack-stage/spack-stage-kokkos-4.6.01-xl7f3yfrba5idkjal25deocxj44elpbk/spack-build-xl7f3yf/core/src -I/tmp/er12izuv/spack-stage/spack-stage-kokkos-4.6.01-xl7f3yfrba5idkjal25deocxj44elpbk/spack-src/core/src -I/tmp/er12izuv/spack-stage/spack-stage-kokkos-4.6.01-xl7f3yfrba5idkjal25deocxj44elpbk/spack-build-xl7f3yf -I/tmp/er12izuv/spack-stage/spack-stage-kokkos-4.6.01-xl7f3yfrba5idkjal25deocxj44elpbk/spack-src/tpls/desul/include -I/tmp/er12izuv/spack-stage/spack-stage-kokkos-4.6.01-xl7f3yfrba5idkjal25deocxj44elpbk/spack-src/tpls/mdspan/include -O3 -DNDEBUG -std=gnu++17 -fPIC -march=core-avx2 -mtune=core-avx2 -mrtm -MD -MT core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_Command_Line_Parsing.cpp.o -MF CMakeFiles/kokkoscore.dir/impl/Kokkos_Command_Line_Parsing.cpp.o.d -o CMakeFiles/kokkoscore.dir/impl/Kokkos_Command_Line_Parsing.cpp.o -c /tmp/er12izuv/spack-stage/spack-stage-kokkos-4.6.01-xl7f3yfrba5idkjal25deocxj44elpbk/spack-src/core/src/impl/Kokkos_Command_Line_Parsing.cpp

```


and  `spack-cc-kokkos-xl7f3yf.out.log`

```
[ccld] /usr/bin/g++-12 -march=skylake -mtune=skylake -L/calculate/kokkos/spack/opt/spack/linux-skylake/gcc-runtime-12.3.0-ezyzimazi56hrxjhlgfohx7rksll6eml/lib -Wl,--disable-new-dtags -Wl,-rpath,/calculate/kokkos/spack/opt/spack/linux-skylake/kokkos-4.6.01-xl7f3yfrba5idkjal25deocxj44elpbk/lib -Wl,-rpath,/calculate/kokkos/spack/opt/spack/linux-skylake/kokkos-4.6.01-xl7f3yfrba5idkjal25deocxj44elpbk/lib64 -Wl,-rpath,/calculate/kokkos/spack/opt/spack/linux-skylake/gcc-runtime-12.3.0-ezyzimazi56hrxjhlgfohx7rksll6eml/lib CMakeCXXCompilerId.cpp
[ccld] /usr/bin/g++-12 -march=skylake -mtune=skylake -L/calculate/kokkos/spack/opt/spack/linux-skylake/gcc-runtime-12.3.0-ezyzimazi56hrxjhlgfohx7rksll6eml/lib -Wl,--disable-new-dtags -Wl,-rpath,/calculate/kokkos/spack/opt/spack/linux-skylake/kokkos-4.6.01-xl7f3yfrba5idkjal25deocxj44elpbk/lib -Wl,-rpath,/calculate/kokkos/spack/opt/spack/linux-skylake/kokkos-4.6.01-xl7f3yfrba5idkjal25deocxj44elpbk/lib64 -Wl,-rpath,/calculate/kokkos/spack/opt/spack/linux-skylake/gcc-runtime-12.3.0-ezyzimazi56hrxjhlgfohx7rksll6eml/lib -print-sysroot
[ld] ld -L/calculate/kokkos/spack/opt/spack/linux-skylake/gcc-runtime-12.3.0-ezyzimazi56hrxjhlgfohx7rksll6eml/lib --disable-new-dtags -rpath /calculate/kokkos/spack/opt/spack/linux-skylake/kokkos-4.6.01-xl7f3yfrba5idkjal25deocxj44elpbk/lib -rpath /calculate/kokkos/spack/opt/spack/linux-skylake/kokkos-4.6.01-xl7f3yfrba5idkjal25deocxj44elpbk/lib64 -rpath /calculate/kokkos/spack/opt/spack/linux-skylake/gcc-runtime-12.3.0-ezyzimazi56hrxjhlgfohx7rksll6eml/lib --help
[vcheck] /usr/bin/g++-12 -std=gnu++17 -v -o CMakeFiles/cmTC_30058.dir/CMakeCXXCompilerABI.cpp.o -c /calculate/kokkos/spack/opt/spack/linux-skylake/cmake-3.31.6-lere57k2rzoie5trdelldw76d3fsa62l/share/cmake-3.31/Modules/CMakeCXXCompilerABI.cpp
[vcheck] /usr/bin/g++-12 -v -Wl,-v CMakeFiles/cmTC_30058.dir/CMakeCXXCompilerABI.cpp.o -o cmTC_30058
[ld] ld -L/calculate/kokkos/spack/opt/spack/linux-skylake/gcc-runtime-12.3.0-ezyzimazi56hrxjhlgfohx7rksll6eml/lib --disable-new-dtags -rpath /calculate/kokkos/spack/opt/spack/linux-skylake/kokkos-4.6.01-xl7f3yfrba5idkjal25deocxj44elpbk/lib -rpath /calculate/kokkos/spack/opt/spack/linux-skylake/kokkos-4.6.01-xl7f3yfrba5idkjal25deocxj44elpbk/lib64 -rpath /calculate/kokkos/spack/opt/spack/linux-skylake/gcc-runtime-12.3.0-ezyzimazi56hrxjhlgfohx7rksll6eml/lib --help
[ld] ld -L/calculate/kokkos/spack/opt/spack/linux-skylake/gcc-runtime-12.3.0-ezyzimazi56hrxjhlgfohx7rksll6eml/lib --disable-new-dtags -rpath /calculate/kokkos/spack/opt/spack/linux-skylake/kokkos-4.6.01-xl7f3yfrba5idkjal25deocxj44elpbk/lib -rpath /calculate/kokkos/spack/opt/spack/linux-skylake/kokkos-4.6.01-xl7f3yfrba5idkjal25deocxj44elpbk/lib64 -rpath /calculate/kokkos/spack/opt/spack/linux-skylake/gcc-runtime-12.3.0-ezyzimazi56hrxjhlgfohx7rksll6eml/lib --help
[cc] /usr/bin/g++-12 -march=skylake -mtune=skylake -std=gnu++17 -o CMakeFiles/cmTC_08b48.dir/cplusplus17.cpp.o -c /tmp/er12izuv/spack-stage/spack-stage-kokkos-4.6.01-xl7f3yfrba5idkjal25deocxj44elpbk/spack-src/cmake/compile_tests/cplusplus17.cpp
[ccld] /usr/bin/g++-12 -march=skylake -mtune=skylake -L/calculate/kokkos/spack/opt/spack/linux-skylake/gcc-runtime-12.3.0-ezyzimazi56hrxjhlgfohx7rksll6eml/lib -Wl,--disable-new-dtags -Wl,-rpath,/calculate/kokkos/spack/opt/spack/linux-skylake/kokkos-4.6.01-xl7f3yfrba5idkjal25deocxj44elpbk/lib -Wl,-rpath,/calculate/kokkos/spack/opt/spack/linux-skylake/kokkos-4.6.01-xl7f3yfrba5idkjal25deocxj44elpbk/lib64 -Wl,-rpath,/calculate/kokkos/spack/opt/spack/linux-skylake/gcc-runtime-12.3.0-ezyzimazi56hrxjhlgfohx7rksll6eml/lib CMakeFiles/cmTC_08b48.dir/cplusplus17.cpp.o -o cmTC_08b48
[ld] ld -L/calculate/kokkos/spack/opt/spack/linux-skylake/gcc-runtime-12.3.0-ezyzimazi56hrxjhlgfohx7rksll6eml/lib --disable-new-dtags -rpath /calculate/kokkos/spack/opt/spack/linux-skylake/kokkos-4.6.01-xl7f3yfrba5idkjal25deocxj44elpbk/lib -rpath /calculate/kokkos/spack/opt/spack/linux-skylake/kokkos-4.6.01-xl7f3yfrba5idkjal25deocxj44elpbk/lib64 -rpath /calculate/kokkos/spack/opt/spack/linux-skylake/gcc-runtime-12.3.0-ezyzimazi56hrxjhlgfohx7rksll6eml/lib --help
[cc] /usr/bin/g++-12 -march=skylake -mtune=skylake -std=gnu++17 -o CMakeFiles/cmTC_1da0e.dir/CheckIncludeFile.cxx.o -c /tmp/er12izuv/spack-stage/spack-stage-kokkos-4.6.01-xl7f3yfrba5idkjal25deocxj44elpbk/spack-build-xl7f3yf/CMakeFiles/CMakeScratch/TryCompile-S1UePn/CheckIncludeFile.cxx
[ld] ld -L/calculate/kokkos/spack/opt/spack/linux-skylake/gcc-runtime-12.3.0-ezyzimazi56hrxjhlgfohx7rksll6eml/lib --disable-new-dtags -rpath /calculate/kokkos/spack/opt/spack/linux-skylake/kokkos-4.6.01-xl7f3yfrba5idkjal25deocxj44elpbk/lib -rpath /calculate/kokkos/spack/opt/spack/linux-skylake/kokkos-4.6.01-xl7f3yfrba5idkjal25deocxj44elpbk/lib64 -rpath /calculate/kokkos/spack/opt/spack/linux-skylake/gcc-runtime-12.3.0-ezyzimazi56hrxjhlgfohx7rksll6eml/lib --help
[cc] /usr/bin/g++-12 -march=skylake -mtune=skylake -std=gnu++17 -o CMakeFiles/cmTC_90bad.dir/CheckIncludeFile.cxx.o -c /tmp/er12izuv/spack-stage/spack-stage-kokkos-4.6.01-xl7f3yfrba5idkjal25deocxj44elpbk/spack-build-xl7f3yf/CMakeFiles/CMakeScratch/TryCompile-tuSJ2G/CheckIncludeFile.cxx
[cc] /usr/bin/g++-12 -march=skylake -mtune=skylake -I/tmp/er12izuv/spack-stage/spack-stage-kokkos-4.6.01-xl7f3yfrba5idkjal25deocxj44elpbk/spack-build-xl7f3yf/simd/src -I/tmp/er12izuv/spack-stage/spack-stage-kokkos-4.6.01-xl7f3yfrba5idkjal25deocxj44elpbk/spack-src/simd/src -I/tmp/er12izuv/spack-stage/spack-stage-kokkos-4.6.01-xl7f3yfrba5idkjal25deocxj44elpbk/spack-build-xl7f3yf -Dkokkossimd_EXPORTS -O3 -DNDEBUG -std=gnu++17 -fPIC -MD -MT simd/src/CMakeFiles/kokkossimd.dir/Kokkos_SIMD_dummy.cpp.o -MF CMakeFiles/kokkossimd.dir/Kokkos_SIMD_dummy.cpp.o.d -o CMakeFiles/kokkossimd.dir/Kokkos_SIMD_dummy.cpp.o -c /tmp/er12izuv/spack-stage/spack-stage-kokkos-4.6.01-xl7f3yfrba5idkjal25deocxj44elpbk/spack-src/simd/src/Kokkos_SIMD_dummy.cpp
[cc] /usr/bin/g++-12 -march=skylake -mtune=skylake -I/tmp/er12izuv/spack-stage/spack-stage-kokkos-4.6.01-xl7f3yfrba5idkjal25deocxj44elpbk/spack-build-xl7f3yf/core/src -I/tmp/er12izuv/spack-stage/spack-stage-kokkos-4.6.01-xl7f3yfrba5idkjal25deocxj44elpbk/spack-src/core/src -I/tmp/er12izuv/spack-stage/spack-stage-kokkos-4.6.01-xl7f3yfrba5idkjal25deocxj44elpbk/spack-build-xl7f3yf -I/tmp/er12izuv/spack-stage/spack-stage-kokkos-4.6.01-xl7f3yfrba5idkjal25deocxj44elpbk/spack-src/tpls/desul/include -I/tmp/er12izuv/spack-stage/spack-stage-kokkos-4.6.01-xl7f3yfrba5idkjal25deocxj44elpbk/spack-src/tpls/mdspan/include -DKOKKOS_DEPENDENCE -Dkokkoscore_EXPORTS -O3 -DNDEBUG -std=gnu++17 -fPIC -march=core-avx2 -mtune=core-avx2 -mrtm -MD -MT core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_Command_Line_Parsing.cpp.o -MF CMakeFiles/kokkoscore.dir/impl/Kokkos_Command_Line_Parsing.cpp.o.d -o CMakeFiles/kokkoscore.dir/impl/Kokkos_Command_Line_Parsing.cpp.o -c /tmp/er12izuv/spack-stage/spack-stage-kokkos-4.6.01-xl7f3yfrba5idkjal25deocxj44elpbk/spack-src/core/src/impl/Kokkos_Command_Line_Parsing.cpp
```

indicate that there are redundant compiler flags `-march=skylake -mtune=skylake ... -march=core-avx2 -mtune=core-avx2` when the sources are build. I believe that `-march=core-avx2 -mtune=core-avx2` wins the race, which results (for my case) in slower code of my application based on deal.ii (@masterleinad).

(i) Why are there redundant compiler flags? 

(ii) Is there a way to enforce only the `-march=skylake -mtune=skylake` flags, without `-march=core-avx2 -mtune=core-avx2`?

### Error message

_No response_

### Information on your system

Output of `spack debug report`
```
$ spack debug report
* **Spack:** 1.0.0.dev0 (0e5cd4b3205e42c0c242792ddf0b1feee3d99bfd)
* **Python:** 3.10.12
* **Platform:** linux-ubuntu22.04-skylake
```

### General information

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have searched the issues of this repo and believe this is not a duplicate
- [x] I have run the failing commands in debug mode and reported the output

@cedricchevalier19 

I attach the content of the relevant *.cmake files located in /lib/cmake/Kokkos. As  I showed in my previous comment, `KokkosTargets.cmake` contains `INTERFACE_COMPILE_OPTIONS "\$<\$<COMPILE_LANGUAGE:CXX>:-march=core-avx2;-mtune=core-avx2;-mrtm>"` which does not match the skylake march flags used be spack. 

Content KokkosConfig.cmake:

```
# No need for policy push/pop. CMake also manages a new entry for scripts
# loaded by include() and find_package() commands except when invoked with
# the NO_POLICY_SCOPE option
# CMP0057 + NEW -> IN_LIST operator in IF(...)
cmake_policy(SET CMP0057 NEW)

# Compute paths

####### Expanded from @PACKAGE_INIT@ by configure_package_config_file() #######
####### Any changes to this file will be overwritten by the next CMake run ####
####### The input file was KokkosConfig.cmake.in                            ########

get_filename_component(PACKAGE_PREFIX_DIR "${CMAKE_CURRENT_LIST_DIR}/../../" ABSOLUTE)

macro(set_and_check _var _file)
  set(${_var} "${_file}")
  if(NOT EXISTS "${_file}")
    message(FATAL_ERROR "File or directory ${_file} referenced by variable ${_var} does not exist !")
  endif()
endmacro()

macro(check_required_components _NAME)
  foreach(comp ${${_NAME}_FIND_COMPONENTS})
    if(NOT ${_NAME}_${comp}_FOUND)
      if(${_NAME}_FIND_REQUIRED_${comp})
        set(${_NAME}_FOUND FALSE)
      endif()
    endif()
  endforeach()
endmacro()

####################################################################################

#Find dependencies
include(CMakeFindDependencyMacro)

#This needs to go above the KokkosTargets in case
#the Kokkos targets depend in some way on the TPL imports


get_filename_component(Kokkos_CMAKE_DIR "${CMAKE_CURRENT_LIST_FILE}" PATH)
include("${Kokkos_CMAKE_DIR}/KokkosTargets.cmake")
include("${Kokkos_CMAKE_DIR}/KokkosConfigCommon.cmake")
unset(Kokkos_CMAKE_DIR)

# check for conflicts
if("launch_compiler" IN_LIST Kokkos_FIND_COMPONENTS AND "separable_compilation" IN_LIST Kokkos_FIND_COMPONENTS)
  message(STATUS "'launch_compiler' implies global redirection of targets depending on Kokkos to appropriate compiler.")
  message(
    STATUS
      "'separable_compilation' implies explicitly defining where redirection occurs via 'kokkos_compilation(PROJECT|TARGET|SOURCE|DIRECTORY ...)'"
  )
  message(FATAL_ERROR "Conflicting COMPONENTS: 'launch_compiler' and 'separable_compilation'")
endif()

if("launch_compiler" IN_LIST Kokkos_FIND_COMPONENTS)
  #
  # if find_package(Kokkos COMPONENTS launch_compiler) then rely on the
  # RULE_LAUNCH_COMPILE and RULE_LAUNCH_LINK to always redirect to the
  # appropriate compiler for Kokkos
  #

  message(
    STATUS
      "kokkos_launch_compiler is enabled globally. C++ compiler commands with -DKOKKOS_DEPENDENCE will be redirected to the appropriate compiler for Kokkos"
  )
  kokkos_compilation(GLOBAL CHECK_CUDA_COMPILES)

elseif(OFF AND NOT CXX STREQUAL CUDA AND NOT "separable_compilation" IN_LIST
                                                                                Kokkos_FIND_COMPONENTS
)
  #
  # if CUDA was enabled, the compilation language was not set to CUDA, and separable compilation was not
  # specified, then set the RULE_LAUNCH_COMPILE and RULE_LAUNCH_LINK globally and
  # kokkos_launch_compiler will re-direct to the compiler used to compile CUDA code during installation.
  # kokkos_launch_compiler will re-direct if ${CMAKE_CXX_COMPILER} and -DKOKKOS_DEPENDENCE is present,
  # otherwise, the original command will be executed
  #

  # run test to see if CMAKE_CXX_COMPILER=nvcc_wrapper
  kokkos_compiler_is_nvcc(IS_NVCC ${CMAKE_CXX_COMPILER})

  # if not nvcc_wrapper and Kokkos_LAUNCH_COMPILER was not set to OFF
  if(NOT IS_NVCC AND (NOT DEFINED Kokkos_LAUNCH_COMPILER OR Kokkos_LAUNCH_COMPILER))
    message(
      STATUS
        "kokkos_launch_compiler is enabled globally. C++ compiler commands with -DKOKKOS_DEPENDENCE will be redirected to the appropriate compiler for Kokkos"
    )
    kokkos_compilation(GLOBAL)
  endif()

  # be mindful of the environment, pollution is bad
  unset(IS_NVCC)
endif()

set(Kokkos_COMPILE_LANGUAGE CXX)
```

Content KokkosConfigCommon.cmake:

```
set(Kokkos_DEVICES SERIAL)
set(Kokkos_OPTIONS DEPRECATED_CODE_4;DEPRECATION_WARNINGS;IMPL_REF_COUNT_BRANCH_UNLIKELY;IMPL_MDSPAN;IMPL_SKIP_COMPILER_MDSPAN;COMPLEX_ALIGN)
set(Kokkos_TPLS LIBDL)
set(Kokkos_ARCH BDW)
set(Kokkos_CXX_COMPILER "/calculate/kokkos/spack/opt/spack/linux-skylake/compiler-wrapper-1.0-gigkxt35gcs2qikmxkrhsmyl7hnonxmo/libexec/spack/gcc/g++")
set(Kokkos_CXX_COMPILER_ID "GNU")
set(Kokkos_CXX_COMPILER_VERSION "12.3.0")
set(Kokkos_CXX_STANDARD 17)

# Required to be a TriBITS-compliant external package
if(NOT TARGET Kokkos::all_libs)
  # CMake Error at <prefix>/lib/cmake/Kokkos/KokkosConfigCommon.cmake:10 (ADD_LIBRARY):
  #   ADD_LIBRARY cannot create ALIAS target "Kokkos::all_libs" because target
  #   "Kokkos::kokkos" is imported but not globally visible.
  if(CMAKE_VERSION VERSION_LESS "3.18")
    set_target_properties(Kokkos::kokkos PROPERTIES IMPORTED_GLOBAL ON)
  endif()
  add_library(Kokkos::all_libs ALIAS Kokkos::kokkos)
endif()

# Export Kokkos_ENABLE_<BACKEND> for each backend that was enabled.
# NOTE: "Devices" is a little bit of a misnomer here.  These are really
# backends, e.g. Kokkos_ENABLE_OPENMP, Kokkos_ENABLE_CUDA, Kokkos_ENABLE_HIP,
# or Kokkos_ENABLE_SYCL.
foreach(DEV ${Kokkos_DEVICES})
  set(Kokkos_ENABLE_${DEV} ON)
endforeach()
# Export relevant Kokkos_ENABLE<OPTION> variables, e.g.
# Kokkos_ENABLE_CUDA_RELOCATABLE_DEVICE_CODE, Kokkos_ENABLE_DEBUG, etc.
foreach(OPT ${Kokkos_OPTIONS})
  set(Kokkos_ENABLE_${OPT} ON)
endforeach()

if(Kokkos_ENABLE_CUDA)
  set(Kokkos_CUDA_ARCHITECTURES )
endif()

if(Kokkos_ENABLE_HIP)
  set(Kokkos_HIP_ARCHITECTURES )
endif()

if(NOT Kokkos_FIND_QUIETLY)
  message(STATUS "Enabled Kokkos devices: ${Kokkos_DEVICES}")
endif()

if(Kokkos_ENABLE_CUDA)
  # If we are building CUDA, we have tricked CMake because we declare a CXX project
  # If the default C++ standard for a given compiler matches the requested
  # standard, then CMake just omits the -std flag in later versions of CMake
  # This breaks CUDA compilation (CUDA compiler can have a different default
  # -std then the underlying host compiler by itself). Setting this variable
  # forces CMake to always add the -std flag even if it thinks it doesn't need it
  set(CMAKE_CXX_STANDARD_DEFAULT 98 CACHE INTERNAL "" FORCE)
endif()

set(KOKKOS_USE_CXX_EXTENSIONS ON)
if(NOT DEFINED CMAKE_CXX_EXTENSIONS OR CMAKE_CXX_EXTENSIONS)
  if(NOT KOKKOS_USE_CXX_EXTENSIONS)
    message(
      WARNING "The installed Kokkos configuration does not support CXX extensions. Forcing -DCMAKE_CXX_EXTENSIONS=Off"
    )
    set(CMAKE_CXX_EXTENSIONS OFF CACHE BOOL "" FORCE)
  endif()
endif()

#   This function makes sure that Kokkos was built with the requested backends
#   and target architectures and generates a fatal error if it was not.
#
#   kokkos_check(
#     [DEVICES <devices>...]   # Set of backends (e.g. "OpenMP" and/or "Cuda")
#     [ARCH <archs>...]        # Target architectures (e.g. "Power9" and/or "Volta70")
#     [OPTIONS <options>...]   # Optional settings (e.g. "TUNING")
#     [TPLS <tpls>...]         # Third party libraries
#     [RETURN_VALUE <result>]  # Set a variable that indicates the result of the
#                              # check instead of a fatal error
#   )
function(kokkos_check)
  set(ALLOWED_ARGS DEVICES ARCH OPTIONS TPLS)
  cmake_parse_arguments(KOKKOS_CHECK "" "RETURN_VALUE" "${ALLOWED_ARGS}" ${ARGN})
  foreach(_arg ${KOKKOS_CHECK_UNPARSED_ARGUMENTS})
    message(SEND_ERROR "Argument '${_arg}' passed to kokkos_check() was not recognized")
  endforeach()
  # Get the list of keywords that were actually passed to the function.
  set(REQUESTED_ARGS)
  foreach(arg ${ALLOWED_ARGS})
    if(KOKKOS_CHECK_${arg})
      list(APPEND REQUESTED_ARGS ${arg})
    endif()
  endforeach()
  set(KOKKOS_CHECK_SUCCESS TRUE)
  foreach(arg ${REQUESTED_ARGS})
    set(MISSING_OPTIONS "")
    foreach(requested ${KOKKOS_CHECK_${arg}})
      set(FOUND_MATCHING_OPTION FALSE)
      foreach(provided ${Kokkos_${arg}})
        string(TOUPPER ${requested} REQUESTED_UC)
        string(TOUPPER ${provided} PROVIDED_UC)
        if(PROVIDED_UC STREQUAL REQUESTED_UC)
          set(FOUND_MATCHING_OPTION TRUE)
        endif()
      endforeach()
      if(NOT FOUND_MATCHING_OPTION)
        list(APPEND MISSING_OPTIONS ${requested})
        set(KOKKOS_CHECK_SUCCESS FALSE)
      endif()
    endforeach()
    if(NOT KOKKOS_CHECK_SUCCESS AND NOT KOKKOS_CHECK_RETURN_VALUE)
      message(STATUS "Could NOT find Kokkos_${arg} (missing: ${MISSING_OPTIONS})")
    endif()
  endforeach()
  if(NOT KOKKOS_CHECK_SUCCESS AND NOT KOKKOS_CHECK_RETURN_VALUE)
    message(FATAL_ERROR "Kokkos does NOT provide all backends and/or architectures requested")
  else()
    set(${KOKKOS_CHECK_RETURN_VALUE} ${KOKKOS_CHECK_SUCCESS} PARENT_SCOPE)
  endif()
endfunction()

# A test to check whether a downstream project set the C++ compiler to NVCC or not
# this is called only when Kokkos was installed with Kokkos_ENABLE_CUDA=ON
function(kokkos_compiler_is_nvcc VAR COMPILER)
  # Check if the compiler is nvcc (which really means nvcc_wrapper).
  execute_process(
    COMMAND ${COMPILER} ${ARGN} --version
    OUTPUT_VARIABLE INTERNAL_COMPILER_VERSION
    OUTPUT_STRIP_TRAILING_WHITESPACE
    RESULT_VARIABLE RET
  )
  # something went wrong
  if(RET GREATER 0)
    set(${VAR} false PARENT_SCOPE)
  else()
    string(REPLACE "\n" " - " INTERNAL_COMPILER_VERSION_ONE_LINE ${INTERNAL_COMPILER_VERSION})
    string(FIND ${INTERNAL_COMPILER_VERSION_ONE_LINE} "nvcc" INTERNAL_COMPILER_VERSION_CONTAINS_NVCC)
    string(REGEX REPLACE "^ +" "" INTERNAL_HAVE_COMPILER_NVCC "${INTERNAL_HAVE_COMPILER_NVCC}")
    if(${INTERNAL_COMPILER_VERSION_CONTAINS_NVCC} GREATER -1)
      set(${VAR} true PARENT_SCOPE)
    else()
      set(${VAR} false PARENT_SCOPE)
    endif()
  endif()
endfunction()

# this function checks whether the current CXX compiler supports building CUDA
function(kokkos_cxx_compiler_cuda_test _VAR _COMPILER)

  file(
    WRITE ${PROJECT_BINARY_DIR}/compile_tests/compiles_cuda.cu
    "
#include <cuda.h>
#include <cstdlib>

__global__
void kernel(int sz, double* data)
{
    int _beg = blockIdx.x * blockDim.x + threadIdx.x;
    for(int i = _beg; i < sz; ++i)
        data[i] += static_cast<double>(i);
}

int main()
{
    double* data = NULL;
    int blocks = 64;
    int grids = 64;
    int ret = cudaMalloc(&data, blocks * grids * sizeof(double));
    if(ret != cudaSuccess)
        return EXIT_FAILURE;
    kernel<<<grids, blocks>>>(blocks * grids, data);
    cudaDeviceSynchronize();
    return EXIT_SUCCESS;
}
"
  )

  # save the command for debugging
  set(_COMMANDS "${_COMPILER} ${ARGN} -c ${PROJECT_BINARY_DIR}/compile_tests/compiles_cuda.cu")

  # use execute_process instead of try compile because we want to set custom compiler
  execute_process(
    COMMAND ${_COMPILER} ${ARGN} -c ${PROJECT_BINARY_DIR}/compile_tests/compiles_cuda.cu
    RESULT_VARIABLE _RET
    WORKING_DIRECTORY ${PROJECT_BINARY_DIR}/compile_tests
    TIMEOUT 15
    OUTPUT_QUIET ERROR_QUIET
  )

  if(NOT _RET EQUAL 0)
    # save the command for debugging
    set(_COMMANDS
        "${_COMMAND}\n${_COMPILER} --cuda-gpu-arch=sm_35 ${ARGN} -c ${PROJECT_BINARY_DIR}/compile_tests/compiles_cuda.cu"
    )
    # try the compile test again with clang arguments
    execute_process(
      COMMAND ${_COMPILER} --cuda-gpu-arch=sm_35 -c ${PROJECT_BINARY_DIR}/compile_tests/compiles_cuda.cu
      RESULT_VARIABLE _RET
      WORKING_DIRECTORY ${PROJECT_BINARY_DIR}/compile_tests
      TIMEOUT 15
      OUTPUT_QUIET ERROR_QUIET
    )
  endif()

  set(${_VAR}_COMMANDS "${_COMMANDS}" PARENT_SCOPE)
  set(${_VAR} ${_RET} PARENT_SCOPE)
endfunction()

# this function is provided to easily select which files use the same compiler as Kokkos
# when it was installed (or nvcc_wrapper):
#
#       GLOBAL      --> all files
#       TARGET      --> all files in a target
#       SOURCE      --> specific source files
#       DIRECTORY   --> all files in directory
#       PROJECT     --> all files/targets in a project/subproject
#
# Use the COMPILER argument to specify a compiler, if needed. By default, it will
# set the values to ${Kokkos_CXX_COMPILER} unless Kokkos_ENABLE_CUDA=ON and
# Kokkos_CXX_COMPILER_ID is NVIDIA, then it will set it to nvcc_wrapper
#
# Use CHECK_CUDA_COMPILES to run a check when CUDA is enabled
#
function(kokkos_compilation)
  cmake_parse_arguments(
    COMP "GLOBAL;PROJECT;CHECK_CUDA_COMPILES" "COMPILER" "DIRECTORY;TARGET;SOURCE;COMMAND_PREFIX" ${ARGN}
  )

  # if built w/o CUDA support, we want to basically make this a no-op
  set(_Kokkos_ENABLE_CUDA OFF)

  if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.17)
    set(MAYBE_CURRENT_INSTALLATION_ROOT "${CMAKE_CURRENT_FUNCTION_LIST_DIR}/../../..")
  endif()

  # search relative first and then absolute
  set(_HINTS "${MAYBE_CURRENT_INSTALLATION_ROOT}" "/calculate/kokkos/spack/opt/spack/linux-skylake/kokkos-4.6.01-xl7f3yfrba5idkjal25deocxj44elpbk")

  # find kokkos_launch_compiler
  find_program(
    Kokkos_COMPILE_LAUNCHER
    NAMES kokkos_launch_compiler
    HINTS ${_HINTS}
    PATHS ${_HINTS}
    PATH_SUFFIXES bin
  )

  if(NOT Kokkos_COMPILE_LAUNCHER)
    message(
      FATAL_ERROR
        "Kokkos could not find 'kokkos_launch_compiler'. Please set '-DKokkos_COMPILE_LAUNCHER=/path/to/launcher'"
    )
  endif()

  # if COMPILER was not specified, assume Kokkos_CXX_COMPILER
  if(NOT COMP_COMPILER)
    set(COMP_COMPILER ${Kokkos_CXX_COMPILER})
    if(_Kokkos_ENABLE_CUDA AND Kokkos_CXX_COMPILER_ID STREQUAL NVIDIA)
      # find nvcc_wrapper
      find_program(
        Kokkos_NVCC_WRAPPER
        NAMES nvcc_wrapper
        HINTS ${_HINTS}
        PATHS ${_HINTS}
        PATH_SUFFIXES bin
      )
      # fatal if we can't nvcc_wrapper
      if(NOT Kokkos_NVCC_WRAPPER)
        message(
          FATAL_ERROR "Kokkos could not find nvcc_wrapper. Please set '-DKokkos_NVCC_WRAPPER=/path/to/nvcc_wrapper'"
        )
      endif()
      set(COMP_COMPILER ${Kokkos_NVCC_WRAPPER})
    endif()
  endif()

  # check that the original compiler still exists!
  if(NOT EXISTS ${COMP_COMPILER})
    message(FATAL_ERROR "Kokkos could not find original compiler: '${COMP_COMPILER}'")
  endif()

  # try to ensure that compiling cuda code works!
  if(_Kokkos_ENABLE_CUDA AND COMP_CHECK_CUDA_COMPILES)

    # this may fail if kokkos_compiler launcher was used during install
    kokkos_cxx_compiler_cuda_test(_COMPILES_CUDA ${Kokkos_COMPILE_LAUNCHER} ${COMP_COMPILER} ${CMAKE_CXX_COMPILER})

    # if above failed, throw an error
    if(NOT _COMPILES_CUDA)
      message(FATAL_ERROR "kokkos_cxx_compiler_cuda_test failed! Test commands:\n${_COMPILES_CUDA_COMMANDS}")
    endif()
  endif()

  if(COMP_COMMAND_PREFIX)
    set(_PREFIX "${COMP_COMMAND_PREFIX}")
    string(REPLACE ";" " " _PREFIX "${COMP_COMMAND_PREFIX}")
    set(Kokkos_COMPILER_LAUNCHER "${_PREFIX} ${Kokkos_COMPILE_LAUNCHER}")
  endif()

  if(COMP_GLOBAL)
    # if global, don't bother setting others
    set_property(
      GLOBAL PROPERTY RULE_LAUNCH_COMPILE "${Kokkos_COMPILE_LAUNCHER} ${COMP_COMPILER} ${CMAKE_CXX_COMPILER}"
    )
    set_property(GLOBAL PROPERTY RULE_LAUNCH_LINK "${Kokkos_COMPILE_LAUNCHER} ${COMP_COMPILER} ${CMAKE_CXX_COMPILER}")
  else()
    foreach(_TYPE PROJECT DIRECTORY TARGET SOURCE)
      # make project/subproject scoping easy, e.g. KokkosCompilation(PROJECT) after project(...)
      if("${_TYPE}" STREQUAL "PROJECT" AND COMP_${_TYPE})
        list(APPEND COMP_DIRECTORY ${PROJECT_SOURCE_DIR})
        unset(COMP_${_TYPE})
      endif()
      # set the properties if defined
      if(COMP_${_TYPE})
        # MESSAGE(STATUS "Using ${COMP_COMPILER} :: ${_TYPE} :: ${COMP_${_TYPE}}")
        set_property(
          ${_TYPE} ${COMP_${_TYPE}} PROPERTY RULE_LAUNCH_COMPILE
                                             "${Kokkos_COMPILE_LAUNCHER} ${COMP_COMPILER} ${CMAKE_CXX_COMPILER}"
        )
        set_property(
          ${_TYPE} ${COMP_${_TYPE}} PROPERTY RULE_LAUNCH_LINK
                                             "${Kokkos_COMPILE_LAUNCHER} ${COMP_COMPILER} ${CMAKE_CXX_COMPILER}"
        )
      endif()
    endforeach()
  endif()
endfunction()
```

Content KokkosTargets.cmake:

```
# Generated by CMake

if("${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION}" LESS 2.8)
   message(FATAL_ERROR "CMake >= 3.0.0 required")
endif()
if(CMAKE_VERSION VERSION_LESS "3.0.0")
   message(FATAL_ERROR "CMake >= 3.0.0 required")
endif()
cmake_policy(PUSH)
cmake_policy(VERSION 3.0.0...3.29)
#----------------------------------------------------------------
# Generated CMake target import file.
#----------------------------------------------------------------

# Commands may need to know the format version.
set(CMAKE_IMPORT_FILE_VERSION 1)

# Protect against multiple inclusion, which would fail when already imported targets are added once more.
set(_cmake_targets_defined "")
set(_cmake_targets_not_defined "")
set(_cmake_expected_targets "")
foreach(_cmake_expected_target IN ITEMS Kokkos::LIBDL Kokkos::kokkoscore Kokkos::kokkoscontainers Kokkos::kokkosalgorithms Kokkos::kokkossimd Kokkos::kokkos)
  list(APPEND _cmake_expected_targets "${_cmake_expected_target}")
  if(TARGET "${_cmake_expected_target}")
    list(APPEND _cmake_targets_defined "${_cmake_expected_target}")
  else()
    list(APPEND _cmake_targets_not_defined "${_cmake_expected_target}")
  endif()
endforeach()
unset(_cmake_expected_target)
if(_cmake_targets_defined STREQUAL _cmake_expected_targets)
  unset(_cmake_targets_defined)
  unset(_cmake_targets_not_defined)
  unset(_cmake_expected_targets)
  unset(CMAKE_IMPORT_FILE_VERSION)
  cmake_policy(POP)
  return()
endif()
if(NOT _cmake_targets_defined STREQUAL "")
  string(REPLACE ";" ", " _cmake_targets_defined_text "${_cmake_targets_defined}")
  string(REPLACE ";" ", " _cmake_targets_not_defined_text "${_cmake_targets_not_defined}")
  message(FATAL_ERROR "Some (but not all) targets in this export set were already defined.\nTargets Defined: ${_cmake_targets_defined_text}\nTargets not yet defined: ${_cmake_targets_not_defined_text}\n")
endif()
unset(_cmake_targets_defined)
unset(_cmake_targets_not_defined)
unset(_cmake_expected_targets)


# Compute the installation prefix relative to this file.
get_filename_component(_IMPORT_PREFIX "${CMAKE_CURRENT_LIST_FILE}" PATH)
get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
if(_IMPORT_PREFIX STREQUAL "/")
  set(_IMPORT_PREFIX "")
endif()

# Create imported target Kokkos::LIBDL
add_library(Kokkos::LIBDL INTERFACE IMPORTED)

set_target_properties(Kokkos::LIBDL PROPERTIES
  INTERFACE_INCLUDE_DIRECTORIES "/usr/include"
  INTERFACE_LINK_LIBRARIES "dl"
)

# Create imported target Kokkos::kokkoscore
add_library(Kokkos::kokkoscore SHARED IMPORTED)

set_target_properties(Kokkos::kokkoscore PROPERTIES
  INTERFACE_COMPILE_DEFINITIONS "\$<\$<COMPILE_LANGUAGE:CXX>:KOKKOS_DEPENDENCE>"
  INTERFACE_COMPILE_FEATURES "cxx_std_17"
  INTERFACE_COMPILE_OPTIONS "\$<\$<COMPILE_LANGUAGE:CXX>:-march=core-avx2;-mtune=core-avx2;-mrtm>"
  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include"
  INTERFACE_LINK_LIBRARIES "Kokkos::LIBDL"
  INTERFACE_LINK_OPTIONS "\$<\$<LINK_LANGUAGE:CXX>:-DKOKKOS_DEPENDENCE;-march=core-avx2;-mtune=core-avx2;-mrtm>"
)

# Create imported target Kokkos::kokkoscontainers
add_library(Kokkos::kokkoscontainers SHARED IMPORTED)

set_target_properties(Kokkos::kokkoscontainers PROPERTIES
  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include"
  INTERFACE_LINK_LIBRARIES "Kokkos::kokkoscore"
)

# Create imported target Kokkos::kokkosalgorithms
add_library(Kokkos::kokkosalgorithms INTERFACE IMPORTED)

set_target_properties(Kokkos::kokkosalgorithms PROPERTIES
  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include"
)

# Create imported target Kokkos::kokkossimd
add_library(Kokkos::kokkossimd SHARED IMPORTED)

set_target_properties(Kokkos::kokkossimd PROPERTIES
  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include"
)

# Create imported target Kokkos::kokkos
add_library(Kokkos::kokkos INTERFACE IMPORTED)

set_target_properties(Kokkos::kokkos PROPERTIES
  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include"
  INTERFACE_LINK_LIBRARIES "Kokkos::kokkoscore;Kokkos::kokkoscontainers;Kokkos::kokkosalgorithms;Kokkos::kokkossimd"
)

# Load information for each installed configuration.
file(GLOB _cmake_config_files "${CMAKE_CURRENT_LIST_DIR}/KokkosTargets-*.cmake")
foreach(_cmake_config_file IN LISTS _cmake_config_files)
  include("${_cmake_config_file}")
endforeach()
unset(_cmake_config_file)
unset(_cmake_config_files)

# Cleanup temporary variables.
set(_IMPORT_PREFIX)

# Loop over all imported files and verify that they actually exist
foreach(_cmake_target IN LISTS _cmake_import_check_targets)
  if(CMAKE_VERSION VERSION_LESS "3.28"
      OR NOT DEFINED _cmake_import_check_xcframework_for_${_cmake_target}
      OR NOT IS_DIRECTORY "${_cmake_import_check_xcframework_for_${_cmake_target}}")
    foreach(_cmake_file IN LISTS "_cmake_import_check_files_for_${_cmake_target}")
      if(NOT EXISTS "${_cmake_file}")
        message(FATAL_ERROR "The imported target \"${_cmake_target}\" references the file
   \"${_cmake_file}\"
but this file does not exist.  Possible reasons include:
* The file was deleted, renamed, or moved to another location.
* An install or uninstall procedure did not complete successfully.
* The installation package was faulty and contained
   \"${CMAKE_CURRENT_LIST_FILE}\"
but not all the files it references.
")
      endif()
    endforeach()
  endif()
  unset(_cmake_file)
  unset("_cmake_import_check_files_for_${_cmake_target}")
endforeach()
unset(_cmake_target)
unset(_cmake_import_check_targets)

# This file does not depend on other imported targets which have
# been exported from the same project but in a separate export set.

# Commands beyond this point should not need to know the version.
set(CMAKE_IMPORT_FILE_VERSION)
cmake_policy(POP)
```

--- a/var/spack/repos/spack_repo/builtin/packages/kokkos/package.py
+++ b/var/spack/repos/spack_repo/builtin/packages/kokkos/package.py
@@ -223,14 +223,10 @@ class Kokkos(CMakePackage, CudaPackage, ROCmPackage):
         "westmere": "WSM",
         "ivybridge": "SNB",
         "broadwell": "BDW",
-        # @AndrewGaspar: Kokkos does not have an arch for plain-skylake - only
-        # for Skylake-X (i.e. Xeon). For now, I'm mapping this to Broadwell
-        # until Kokkos learns to optimize for SkyLake without the AVX-512
-        # extensions. SkyLake with AVX-512 will still be optimized using the
-        # separate `skylake_avx512` arch.
-        "skylake": "BDW",
-        "icelake": "SKX",
+        "skylake": "SKL",
+        "icelake": "ICL",
         "skylake_avx512": "SKX",
+        "sapphirerapids": "SPR",
     }
 
     spack_cuda_arch_map = {

